resteasy-spring-boot-starter/src/main/java/com/paypal/springboot/resteasy/ResteasySpringBootConfig.java

```diff
@@ -35,17 +35,15 @@
 
 	private static Logger logger = LoggerFactory.getLogger(ResteasySpringBootConfig.class);
 
-	private static ResteasyProviderFactory resteasyProviderFactory = new ResteasyProviderFactory();
```

Comment:
This Spring bean is a singleton, so there is no need to change the scope of resteasyProviderFactory and resourceMethodRegistry. In the end of the day, the change you did in method springBeanProcessor makes no difference. Method springBeanProcessor is called only once (there will be only one bean of type BeanFactoryPostProcessor created by springBeanProcessor).